Yona/Yona/Others/SMSValidPasscodeLogin/Login/LoginViewController.swift

```diff
@@ -71,8 +73,13 @@ extension LoginViewController: CodeInputViewDelegate {
                         self.navigationController?.pushViewController(passcode, animated: false)
                         }
                     } else {
-                        let storyboard = UIStoryboard(name: "Dashboard", bundle: NSBundle.mainBundle())
-                        self.view.window?.rootViewController = storyboard.instantiateInitialViewController()
+                        let appDelegate = UIApplication.sharedApplication().delegate as! AppDelegate
+                        let aVariable = appDelegate.viewControllers
+                        if self.view.window?.rootViewController == aVariable {
```

Comment:
1- We should not have an array of all view controllers in the app delegate
2- What kind of naming is "aVariable"? beside the name, it is a singular that should hold array of view controllers, which are plural :)